src/libappimage/desktop_integration/integrator/Integrator.cpp

```diff
@@ -225,8 +224,8 @@ namespace appimage {
                         auto deployPath = generateDeployPath(iconPath);
                         icon.save(deployPath.string(), icon.format());
                     } catch (const utils::IconHandleError& er) {
-                        logger.error() << er.what() << std::endl;
-                        logger.error() << "No icon was generated for: " << appImage.getPath() << std::endl;
+                        Logger::error(er.what());
```

Comment:
There should not be two calls. How about a method `Logger::exception(const std::exception& e, ...)`? You could overload the method to auto-call `what()` if possible (or maybe use templates and type traits).

Reply:
Or a `LOG_EXCEPTION` macro that _automatically_ evaluates `__FUNCTION__`.